### PR TITLE
Allow turning off jobs in a zone, optionally per application

### PR DESCRIPTION
**Pull request by jonmv**
@freva please review and mereg.

### CODE REVIEW

**Comment by jonmv** [comment]
Could consider wiring this through `DeploymentStatus` alongside `coolingDownUntil` et al to show, if we end up using it much ... but it's not straightforward to get it all right.

File: flags/src/main/java/com/yahoo/vespa/flags/PermanentFlags.java [hunk 1]
```diff
@@ -296,6 +296,13 @@ public class PermanentFlags {
             "Takes effect at redeployment",
             ZONE_ID, APPLICATION_ID);
 
+    public static final UnboundListFlag<String> DISABLED_DEPLOYMENT_ZONES = defineListFlag(
```
Comment (freva) [inline flags/src/main/java/com/yahoo/vespa/flags/PermanentFlags.java:299]: Why not boolean flag with `ZONE_ID` dimension?

Comment (jonmv) [reply flags/src/main/java/com/yahoo/vespa/flags/PermanentFlags.java:299]: AFAIK that zone dimension is the zone of the requestor, but that's always the controller zone in this case?

Comment (jonmv) [reply flags/src/main/java/com/yahoo/vespa/flags/PermanentFlags.java:299]: Ah, maybe that's not the case ... Then that's obviously the right thing.

**Review by freva — APPROVED** [review]

**Pull request merged**